lib/index.d.ts

```diff
@@ -0,0 +1,7 @@
+import {ParserOptions} from './parse'
+
+declare function buffer(value: string | Buffer, options?: ParserOptions): string
+
+declare function buffer(value: string | Buffer, encoding?: BufferEncoding, options?: ParserOptions): string
```

Comment:
> Is `BufferEncoding` injected by TS because this is defined as a Node module?

Yes. `BufferEncoding` is from `node_modules/@types/node/` 

> If so: in `vfile` we opted to copy the list instead:

It easy to copy `BufferEncoding`, but it's not easy to copy [`Buffer`](https://github.com/ocavue/micromark/blob/6a53e9486c0ba1e9faed3d70ebc03f162d410f87/lib/index.d.ts#L6), which is also from `@types/node` but more complex. Maybe we should add `"@types/node": ">=10.0.0"` as one of the dependencies?